ros/processor/insights_engine_result_consumer.py

```diff
@@ -115,7 +128,42 @@ def process_report(self, host, reports):
                     fqdn=host['fqdn'],
                     rule_hit_details=reports,
                     number_of_recommendations=rec_count,
-                    state=SYSTEM_STATES[state_key]
+                    state=SYSTEM_STATES[state_key],
+                    instance_type=performance_record.get('instance_type')
+                )
+                LOG.info(
+                    f"{self.prefix} - System created/updated successfully: {host['id']}"
+                )
+
+                # For Optimized state, reports would be empty, but utilization_info would be present
+                if reports:
+                    set_default_utilization = True if reports[0].get('key') == 'NO_PCP_DATA' else False
+                else:
+                    set_default_utilization = False
+
+                if set_default_utilization is False:
```

Comment:
Can we check True condition first, if there is no specific reason for checking False first?
```suggestion
                if set_default_utilization:
```

Reply:
The True will only apply for 'NO_PCP_DATA', hence if we evaluate False first then the processor code will get executed faster IMO.

Reply:
+1
minor fix:
```suggestion
                if not set_default_utilization:
```

Reply:
I might be missing the point here, but I feel `is` is more explicit, clearer and good for checking booleans.
`not` feels a little confusing for me cause of the following,

`>>> not False`
True
`>>> not 0` :thinking: 
True
`>>> not 1`
False
`>>> not {}` :thinking: 
True


Please do correct me if I'm wrong

Reply:
Let's keep it as is for now.